pkg/verification/verification_service_test.go

```diff
@@ -150,6 +159,96 @@ func (s *TestVerificationServiceSuite) TestSendVerifyMessageFails() {
 	require.Empty(s.T(), userSignup.Annotations[v1alpha1.UserSignupVerificationCodeAnnotationKey])
 }
 
+func (s *TestVerificationServiceSuite) TestVerifyCode() {
```

Comment:
could we also add tests for the other error conditions? like if the attemps field and the expiry field are corrupt/set incorrectly/not set at all

Reply:
I've added a bunch of additional tests, I think all scenarios are now covered.